src/Tools/dotnet-counters/CounterProvider.cs

```diff
@@ -40,12 +43,12 @@ public static string SerializeUnknownProviderName(string unknownCounterProviderN
         }
 
         public IReadOnlyList<CounterProfile> GetAllCounters() => Counters.Values.ToList();
-
     }
 
     public class CounterProfile
     {
         public string Name { get; set; }
         public string Description { get; set; }
+        public string[] SupportedVersions { get; set; }
```

Comment:
Could we use an enum here instead so we don't have magic strings for version filtering?  Or at least a single definition of the string that represents each version so we don't have many different instances of the same string value?  Then it's just a matter of normalizing the entered version to the runtime, which we should probably do regardless of whether this is a set of strings or enums.  As it is, if a user doesn't enter precisely `"3.1"`, and instead enters `"3.1.7-1234.6"`, then it will fail your checks here.

Reply:
Just came back to this PR - I don't think changing to enum makes much sense after I thought about it a little.... 

How about I update the help text to do something like this instead?
```
(tf2) ➜  dotnet-counters git:(dev/suwhang/1368) ✗ dotnet run list --runtime-version 3.1.7-1234.6
3.1.7-1234.6 is not a supported version string or a supported runtime version.
Supported version strings: 3.0, 3.1, 5.0
```

Reply:
I think that's fine.  If you don't do the enum, could you still make all the uses of `"5.0"`, `"3.0"`, and `"3.1"` refer to constant variables?  For example, 
```cs
static private readonly net50 = "5.0";
static private readonly net31 = "3.1";
static private readonly net30 = "3.0";
```

That way we don't have many definitions of the same string with the same meaning.